src/auth/basic/SSPI/basic_sspi_auth.cc

```diff
@@ -149,7 +151,23 @@ main(int argc, char **argv)
         /* Clear any current settings */
         username[0] = '\0';
         password[0] = '\0';
-        sscanf(wstr, "%s %s", username, password);  /* Extract parameters */
+
+        parsed = sscanf(wstr, " %255s%1c%255s %c", username, &sep, password, &junk);
+
+        if (parsed < 2 || !isspace((unsigned char)sep)) {
+            username[0] = '\0';
+            password[0] = '\0';
+            puts("ERR");
```

Comment:
This puts() call appears to be a poor duplication of existing SEND_ERR()-based error handling code further below. Please refactor to avoid code duplication, especially poor one.